flink-connectors/flink-sql-connector-hbase-1.4/pom.xml

```diff
@@ -87,6 +87,7 @@ under the License.
 									<exclude>org.apache.hbase:hbase-metrics*</exclude>
 									<exclude>org.apache.hbase:hbase-server*</exclude>
 									<exclude>org.apache.hbase:hbase-hadoop*-compat</exclude>
+									<exclude>org.apache.hbase:hbase-common:jar:tests</exclude>
```

Comment:
hmm?

Reply:
```
[INFO] --- maven-shade-plugin:3.1.1:shade (shade-flink) @ flink-sql-connector-hbase-1.4_2.11 ---
[INFO] Including org.apache.flink:flink-connector-hbase-1.4_2.11:jar:1.12-SNAPSHOT in the shaded jar.
[INFO] Including org.apache.flink:flink-connector-hbase-base_2.11:jar:1.12-SNAPSHOT in the shaded jar.
[INFO] Excluding org.apache.hbase:hbase-server:jar:1.4.3 from the shaded jar.
[INFO] Including org.apache.hbase:hbase-common:jar:1.4.3 in the shaded jar.
[INFO] Excluding org.apache.avro:avro:jar:1.10.0 from the shaded jar.
[INFO] Excluding com.fasterxml.jackson.core:jackson-core:jar:2.10.1 from the shaded jar.
[INFO] Excluding com.fasterxml.jackson.core:jackson-databind:jar:2.10.1 from the shaded jar.
[INFO] Excluding com.fasterxml.jackson.core:jackson-annotations:jar:2.10.1 from the shaded jar.
[INFO] Excluding com.github.stephenc.findbugs:findbugs-annotations:jar:1.3.9-1 from the shaded jar.
[INFO] Including org.apache.hbase:hbase-protocol:jar:1.4.3 in the shaded jar.
[INFO] Including org.apache.hbase:hbase-procedure:jar:1.4.3 in the shaded jar.
[INFO] Including org.apache.hbase:hbase-common:jar:tests:1.4.3 in the shaded jar.
```

I believe these are test classes only, which we don't need to ship to our users?

Reply:
If they are only test classes then I'm wondering why this shows up in the dependency tree, and an exclusion or explicit test dependency in the connector module might be more appropriate.

Reply:
Yes, true. I will look into this.

Reply:
This is the dependency tree:
```
[INFO] org.apache.flink:flink-sql-connector-hbase-1.4_2.11:jar:1.12-SNAPSHOT
[INFO] +- org.apache.flink:flink-connector-hbase-1.4_2.11:jar:1.12-SNAPSHOT:compile
[INFO] |  \- org.apache.flink:flink-connector-hbase-base_2.11:jar:1.12-SNAPSHOT:compile
[INFO] |     +- org.apache.hbase:hbase-server:jar:1.4.3:compile
[INFO] |     |  +- org.apache.hbase:hbase-common:jar:1.4.3:compile
[INFO] |     |  |  +- org.apache.avro:avro:jar:1.10.0:compile
[INFO] |     |  |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.10.1:compile
[INFO] |     |  |  |  \- com.fasterxml.jackson.core:jackson-databind:jar:2.10.1:compile
[INFO] |     |  |  |     \- com.fasterxml.jackson.core:jackson-annotations:jar:2.10.1:compile
[INFO] |     |  |  \- com.github.stephenc.findbugs:findbugs-annotations:jar:1.3.9-1:compile
[INFO] |     |  +- org.apache.hbase:hbase-protocol:jar:1.4.3:compile
[INFO] |     |  +- org.apache.hbase:hbase-procedure:jar:1.4.3:compile
[INFO] |     |  |  \- org.apache.hbase:hbase-common:jar:tests:1.4.3:compile
```

I can not add an exclude on hbase-server, because I can not exclude based on the `tests` classifier.

The other question is: why are we depending on `hbase-server`. It would probably be fine to depend on `hbase-common` and `hbase-client` (just judging from the names). I quickly tried that, but it lead to all kinds of problems with the tests.
How about working with the shade exclude for now, but filing a ticket, to solve this properly in the next release cycle?

Reply:
What about declaring it as a test dependency? I checked the hbase-procedure module, and it doesn't actually seem to require it outside of tests.
```
<dependency>
	<groupId>org.apache.hbase</groupId>
	<artifactId>hbase-common</artifactId>
	<version>${hbase.version}</version>
	<classifier>tests</classifier>
	<scope>test</scope>
</dependency>
```

Reply:
As for whether we should depend on hbase-server, it may make sense to look into it.

Reply:
I'll try the test dependency approach, and I've filed a ticket: https://issues.apache.org/jira/browse/FLINK-19971